### PR TITLE
OS 35 - Added Elasticsearch as jar to registry

### PR DESCRIPTION
**Pull request by Hari-stackroute**
1.implemented add docoument with id for elasticsearch
2.created elastic-search project and added as a jar to registry
3.got a issue when elastic-search added as jar
    Issue is version of elastic-search defaultly picking as 5.6.4 which is configured for spring boot, it is overrided.

### CODE REVIEW

File: java/registry/src/main/java/io/opensaber/registry/dao/RegistryDaoImpl.java [hunk 1]
```diff
@@ -53,6 +64,7 @@ public DatabaseProvider getDatabaseProvider() {
     public String addEntity(Graph graph, JsonNode rootNode) {
         VertexWriter vertexWriter = new VertexWriter(graph, getDatabaseProvider(), uuidPropertyName);
         String entityId = vertexWriter.writeNodeEntity(rootNode);
+        elasticService.addEntity("os-teacher", entityId, JSONUtil.convertJsonNodeToMap(rootNode));
```
Comment (indrajra) [inline java/registry/src/main/java/io/opensaber/registry/dao/RegistryDaoImpl.java]: why is dao doing this? It must be service (not dao).

File: java/elastic-search/pom.xml [hunk 2]
```diff
@@ -0,0 +1,101 @@
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <groupId>io.opensaber</groupId>
+    <artifactId>elastic-search</artifactId>
+    <version>2.0.1</version>
+    <name>Elastic-Search</name>
+    <description>Elastic Search for OS</description>
+
+    <parent>
```
[the remark this answers is not in the document]
Comment (Hari-stackroute) [reply java/elastic-search/pom.xml]: removed.

File: java/registry/src/main/resources/application.yml.sample [hunk 3]
```diff
@@ -170,6 +170,16 @@ taskExecutor:
     # Set the capacity for the ThreadPoolExecutor's BlockingQueue.
     queueCapacity: ${taskExecutor_index_queueCapacity:3}
 
+# elastic-search configuration details
+elastic:
+  search:
+    # elastic-search index name
```
[the remark this answers is not in the document]
Comment (Hari-stackroute) [reply java/registry/src/main/resources/application.yml.sample]: yes removed from config file and taken care in code. loading indices from definition manager and created indicies.

File: java/registry/src/main/java/io/opensaber/registry/dao/RegistryDaoImpl.java (hunk 1 above)
Comment (Hari-stackroute) [reply java/registry/src/main/java/io/opensaber/registry/dao/RegistryDaoImpl.java]: Aggred, it should not be here so moved to reg-service layer.

**Pull request merged**